src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,35 @@
+package seedu.addressbook.data.person;
+
+public class Contact {
+    public String value;
+    private boolean isPrivate;
+
+    public Contact(boolean isPrivate) {
+        this.isPrivate = isPrivate;
+    }
+
+    public void setValue(String value) {
+        this.value = value;
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public int hashCode() {
+        return value.hashCode();
+    }
+
+    public boolean isPrivate() {
+        return isPrivate;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
This is fine, but note that now an `Email` and `Address` having the same value will be considered equal even if they are of different types.